prime-angular-frontend/src/app/modules/site-registration/pages/hours-operation/hours-operation.component.ts

```diff
@@ -52,30 +72,31 @@ export class HoursOperationComponent implements OnInit, IPage, IForm {
 
   public onSubmit() {
     if (this.formUtilsService.checkValidity(this.businessDays)) {
-      this.hasNoHours = false;
-
       const payload = this.siteFormStateService.json;
       this.siteResource
         .updateSite(payload)
         .subscribe(() => {
           this.form.markAsPristine();
           this.nextRoute();
         });
-    } else {
-      this.hasNoHours = true;
     }
   }
 
-  public onAdd(businessDay: BusinessDay[]) {
-    this.hasNoHours = false;
+  public getWeekDay(num: number) {
+    return WeekDay[num];
+  }
 
-    this.formUtilsService.formArrayPush(this.businessDays, businessDay);
-    const sorted = this.businessDays.value.sort(this.sortConfigByDay());
-    this.businessDays.patchValue(sorted);
+  public hasDay(index: number): boolean {
+    return (this.businessDays.value[index].startTime !== null) ? true : false;
```

Comment:
We could pass in the group instead of the index, and check the value.  Something like `return !!group.get('startTime').value;`